From 5c15bfb0e9203b6fac1eec628cce9c80e6603e13 Mon Sep 17 00:00:00 2001
From: Felix Radensky <felix.r@variscite.com>
Date: Wed, 19 May 2021 18:25:54 +0300
Subject: [PATCH 10/12] imx8mm_var_dart: Set console environment variable in
 the default environment

Instead of defining the console environment variable in the source code
do it in the default environment.
---
 board/variscite/imx8mm_var_dart/imx8mm_var_dart.c |  1 -
 include/configs/imx8mm_var_dart.h                 | 14 +++++++++++---
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/board/variscite/imx8mm_var_dart/imx8mm_var_dart.c b/board/variscite/imx8mm_var_dart/imx8mm_var_dart.c
index 38c34bef97..b78c675058 100644
--- a/board/variscite/imx8mm_var_dart/imx8mm_var_dart.c
+++ b/board/variscite/imx8mm_var_dart/imx8mm_var_dart.c
@@ -209,7 +209,6 @@ int board_late_init(void)
 
 	if (id == VAR_SOM_MX8M_MINI) {
 		env_set("board_name", "VAR-SOM-MX8M-MINI");
-		env_set("console", "ttymxc3,115200");
 		switch (som_rev) {
 		case SOM_REV_10:
 			env_set("som_rev", "som_rev10");
diff --git a/include/configs/imx8mm_var_dart.h b/include/configs/imx8mm_var_dart.h
index 3fa6db9ddb..0045663f33 100644
--- a/include/configs/imx8mm_var_dart.h
+++ b/include/configs/imx8mm_var_dart.h
@@ -67,7 +67,7 @@
 	"bootdir=/boot\0"	\
 	"script=boot.scr\0" \
 	"image=Image.gz\0" \
-	"console=ttymxc0,115200\0" \
+	"console=undefined\0" \
 	"img_addr=0x42000000\0"			\
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
@@ -92,7 +92,15 @@
 		"fi; " \
 		"bootaux ${m4_addr};\0" \
 	"optargs=setenv bootargs ${bootargs} ${kernelargs};\0" \
-	"mmcargs=setenv bootargs console=${console} " \
+	"setconsole=" \
+		"if test $console = undefined; then " \
+			"if test $board_name = VAR-SOM-MX8M-MINI; then " \
+				"setenv console ttymxc3,115200; " \
+			"else " \
+				"setenv console ttymxc0,115200; " \
+			"fi; " \
+		"fi; \0" \
+	"mmcargs=run setconsole; setenv bootargs console=${console} " \
 		"root=/dev/mmcblk${mmcblk}p${mmcpart} rootwait rw ${cma_size}\0 " \
 	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootdir}/${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
@@ -136,7 +144,7 @@
 		"else " \
 			"echo wait for boot; " \
 		"fi;\0" \
-	"netargs=setenv bootargs console=${console} " \
+	"netargs=run setconsole; setenv bootargs console=${console} " \
 		"root=/dev/nfs ${cma_size} " \
 		"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0" \
 	"netboot=echo Booting from net ...; " \
-- 
2.35.1

